### PR TITLE
patron profile: show overdue preview fees

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Closes rero/rero-ils#3343.

## Display

patron-profile

### CODE REVIEW

**Review by PascalRepond — APPROVED** [review]

File: projects/public-search/src/app/patron-profile/patron-profile-loans/patron-profile-loan/patron-profile-loan.component.ts [hunk 1]
```diff
@@ -67,19 +76,29 @@ export class PatronProfileLoanComponent implements OnInit {
...
+    private _patronProfileMenuService: PatronProfileMenuService,
+    private _patronProfileService: PatronProfileService
   ) {}
 
   /** OnInit hook */
   ngOnInit(): void {
     this._loanApiService
       .canExtend(this.record.metadata.pid)
       .subscribe((response: CanExtend) => this.canExtend = response);
+    if (this.record.metadata.is_late) {
```
Comment (zannkukai) [inline projects/public-search/src/app/patron-profile/patron-profile-loans/patron-profile-loan/patron-profile-loan.component.ts:94]: you should have a `.is_overdue` flag that could be more relevant in this case.

`is_late` : is enddate < currentTimestamp
`is_overdue`: is enddate < currrentTimestamp && relatedCipo.firstFeeStep <= currentTimestamp

By exemple, a cipo can define a 2 days "pardon" period. 
If patron return the book at J+1 : `is_late=true` and `is_overdue=false`
If patron return the book at J+3 : `is_late=true` and `is_overdue=true`

Comment (Garfield-fr) [reply projects/public-search/src/app/patron-profile/patron-profile-loans/patron-profile-loan/patron-profile-loan.component.ts:94]: The flag was not called `is_overdue`, but simply `overdue` 😉. I've changed that.

File: projects/public-search/src/app/patron-profile/patron-profile-loans/patron-profile-loan/patron-profile-loan.component.html [hunk 2]
```diff
@@ -47,7 +47,12 @@
           {{ record.metadata.extension_count }}
           {{ record.metadata.extension_count | i18nPlural: {'=1': 'renewal', 'other': 'renewals'} | translate }}
         </div>
-        <div *ngIf="record.metadata.is_late" class="badge badge-warning font-weight-normal" translate>overdue</div>
+        <div *ngIf="record.metadata.is_late">
```
Comment (zannkukai) [inline projects/public-search/src/app/patron-profile/patron-profile-loans/patron-profile-loan/patron-profile-loan.component.html:50]: `<div *ngIf="record.metadata.overdue">` ?

File: projects/public-search/src/app/patron-profile/patron-profile.component.ts [hunk 3]
```diff
@@ -110,6 +113,10 @@ export class PatronProfileComponent implements OnInit, OnDestroy {
 
   /** OnInit hook */
   ngOnInit(): void {
+    this.feeTotal = 0;
+    this._subscription.add(
+      this._patronProfileService.loanFeesEvent$.subscribe((fees: number) => this.loansFeesTotal += fees)
```
Comment (zannkukai) [inline projects/public-search/src/app/patron-profile/patron-profile.component.ts:118]: Javascript and floating number are problematic... you need to round the result with 2 decimals. See this simple exemple :
```javascript
let total = 0.01;
console.log(total);
for (let idx=0; idx<10; idx++) {
 total += 0.01;
 console.log(total);
}
```
result
```
0.01
0.02
0.03
0.04
0.05
0.060000000000000005
0.07
0.08
0.09
0.09999999999999999
0.10999999999999999
```

**Review by zannkukai — CHANGES_REQUESTED** [review]

**Review by zannkukai — APPROVED** [review]

File: projects/public-search/src/app/api/loan-api.service.ts [hunk 4]
```diff
@@ -124,6 +125,15 @@ export class LoanApiService extends BaseApi {
       );
   }
 
+  /**
+   * Get the preview overdue fees related to a specific loan
+   * @param loanPid - string: the Loan pid
+   */
+  getPreviewOverdue(loanPid: string): Observable<LoanOverduePreview> {
+    const url = `/api/loan/${loanPid}/overdue/preview`;
```
Comment (jma) [inline projects/public-search/src/app/api/loan-api.service.ts:133]: Can you use the ng-core service to get the prefix of the `/api` url?

**Review by jma — APPROVED** [review]

**Pull request merged**